general/array.hpp

```diff
@@ -148,6 +148,9 @@ class Array : public BaseArray
    /// Return the first index where 'el' is found; return -1 if not found
    inline int Find(const T &el) const;
 
+   /// Do bisection search for 'el' in a sorted array; return -1 if not found.
+   inline int FindSorted(const T &el) const;
```

Comment:
Should we call this method `LowerBound`? The function `std::lower_bound`, used in the implementation, is more general, so the name should reflect that.

Reply:
I think we should not, because:
1. FindSorted it is not as general as std::lower_bound, which always returns a bound even if the given element does not exist,
2. FindSorted is a companion to Find in the interface,
3. We could have used something else than std::lower_bound in the implementation, like bsearch from stdlib.h, the actual implementation is not that important IMO.

Reply:
OK, let's keep it as is.